bindingcollectionadapter-paging/src/main/java/me/tatarka/bindingcollectionadapter2/itembindings/OnItemBindLoadState.java

```diff
@@ -0,0 +1,103 @@
+            }
+            if (loadState == null && headerLoadState == null && footerLoadState == null) {
+                throw new IllegalStateException("loadState itemBind must not be null");
+            }
+            if (loadState == null) {
+                loadState = new OnItemBind<Load>() {
+                    @Override
+                    public void onItemBind(@NonNull ItemBinding itemBinding, int position, Load item) {
+                        switch (item.getLoadType()) {
+                            case REFRESH:
+                                //TODO
+                                break;
+                            case PREPEND:
+                                headerLoadState.onItemBind(itemBinding, position, item.getLoadState());
+                                break;
+                            case APPEND:
+                                footerLoadState.onItemBind(itemBinding, position, item.getLoadState());
+                                break;
+                        }
+                    }
+                };
+            }
+            return new OnItemBindLoadState<>(itemBind, loadState);
+        }
+
+        public Builder<T> item(int variableId, @LayoutRes int layoutRes) {
+            this.itemBind = itemBind(variableId, layoutRes);
+            return this;
+        }
+
+        public Builder<T> item(@NonNull OnItemBind<T> onItemBind) {
+            this.itemBind = onItemBind;
+            return this;
+        }
+
+        public Builder<T> headerLoadState(@NonNull OnItemBind<androidx.paging.LoadState> onItemBind) {
+            this.headerLoadState = onItemBind;
+            return this;
+        }
+
+        public Builder<T> headerLoadState(int variableId, @LayoutRes int layoutRes) {
+            this.headerLoadState = itemBind(variableId, layoutRes);
+            return this;
+        }
+
+        public Builder<T> footerLoadState(@NonNull OnItemBind<androidx.paging.LoadState> onItemBind) {
+            this.footerLoadState = onItemBind;
+            return this;
+        }
+
+        public Builder<T> footerLoadState(int variableId, @LayoutRes int layoutRes) {
+            this.footerLoadState = itemBind(variableId, layoutRes);
+            return this;
+        }
+
+        @NonNull
+        private static <T> OnItemBind<T> itemBind(final int variableId, @LayoutRes final int layoutRes) {
+            return new OnItemBind<T>() {
+                @Override
+                public void onItemBind(@NonNull ItemBinding itemBinding, int position, T item) {
+                    itemBinding.set(variableId, layoutRes);
+                }
+            };
+        }
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void onItemBind(@NonNull ItemBinding itemBinding, int position, Object item) {
+        if (item instanceof Load) {
+            combinedLoadStatesItemBind.onItemBind(itemBinding, position, (Load) item);
```

Comment:
Could we plug in the list reference here as an extra binding?  That way we could have the retry callback support.  We could have consumer pass in variable id in OnItemBindLoadState constructor.

Reply:
I was thinking

```
@Override
    @SuppressWarnings("unchecked")
    public void onItemBind(@NonNull ItemBinding itemBinding, int position, Object item) {
        if (item instanceof Load) {
            itemBinding.bindExtra(pagingCallbackVariableId, pagedDataCallback);
            combinedLoadStatesItemBind.onItemBind(itemBinding, position, (Load) item);
        } else {
            itemBind.onItemBind(itemBinding, position, (T) item);
        }
    }
```